docs/templating.md

```diff
@@ -892,41 +892,816 @@ otherwise.
+
+**Input**
+
+```jinja
+{% set letters = aaabbbccc%}
+{{ "letters" | replace("", ".") }}
+```
+
+**Output**
+
+```jinja
+.l.e.t.t.e.r.s.
+
+```
+
+Every instance of an item up to a given number (item to be replaced,
+item replacement, number to be replaced):
+
+**Input**
+
+```jinja
+{% set letters = "aaabbbccc" %}
+{{ letters | replace("a", "x", 2) }}
+```
+Note in this instance the required quote marks surrounding the list.
+
+**Output**
+
+```jinja
+xxabbbccc
+```
+
+It is possible to search for patterns in a list to replace:
+
+**Input**
+
+```jinja
+{% set letters = "aaabbbccc" %}
+{{ letters | replace("ab", "x", 2) }}
+```
+
+**Output**
+
+```jinja
+aaxbbccc
+```
+
+### reverse
+
+Reverse a string:
+
+**Input**
+
+```jinja
+{{ "abcdef" | reverse }}
+```
+
+**Output**
+
+```jinja
+fedcba
+```
+
+Reverse an array:
+
+**Input**
+
+```jinja
+{% for i in [1, 2, 3, 4] | reverse %}
+    {{ i }}
+{% endfor %}
+```
+
+**Output**
+
+```jinja
+4 3 2 1
+```
+
+### round
+
+Round a number:
+
+**Input**
+
+```jinja
+{{ 4.5 | round }}
+```
+
+**Output**
+
+```jinja
+5
+```
+
+Round to the nearest whole number (which rounds down):
+
+**Input**
+
+```jinja
+{{ 4 | round(0, "floor")
+```
+
+**Output**
+
+```jinja
+4
+```
+
+Specify the number of  digits to round:
+
+**Input**
+
+```jinja
+{{ 4.12346 | round(4) }}
+```
+
+**Output**
+
+```jinja
+4.1235
+```
+
+### safe
+
+Mark the value as safe which means that in an environment with automatic
+escaping enabled this variable will not be escaped.
+
+**Input**
+
+```jinja
+{{ "foo http://www.example.com/ bar" | urlize | safe }}
+```
+
+**Output**
+
+```jinja
+foo <a href="http://www.example.com/">http://www.example.com/</a> bar
+```
+
+### selectattr (only the single-argument form)
+
+This is the opposite to ```rejectattr```.
+
+Filters a sequence of objects by applying a test to the specified attribute
+of each object, and only selecting the objects with the test succeeding.
+
+If no test is specified, the attribute’s value will be evaluated as a boolean.
+
+**Input**
+
+```jinja
+{% set foods = [{tasty: true}, {tasty: false}, {tasty: true}]%}
+{{ foods | selectattr("tasty") | length }}
+```
+
+**Output**
+
+```jinja
+2
+```
+
+### slice
+
+Slice an iterator and return a list of lists containing those items:
+
+**Input**
+
+```jinja
+{% set arr = [1,2,3,4,5,6,7,8,9] %}
+
+<div class="columwrapper">
+  {%- for items in arr | slice(3) %}
+    <ul class="column-{{ loop.index }}">
+    {%- for item in items %}
+      <li>{{ item }}</li>
+    {%- endfor %}
+    </ul>
+  {%- endfor %}
+</div>
+```
+
+**Output**
+
+```jinja
+<div class="columwrapper">
+    <ul class="column-1">
+      <li>1</li>
+      <li>2</li>
+      <li>3</li>
+    </ul>
+    <ul class="column-2">
+      <li>4</li>
+      <li>5</li>
+      <li>6</li>
+    </ul>
+    <ul class="column-3">
+      <li>7</li>
+      <li>8</li>
+      <li>9</li>
+    </ul>
+</div>
+```
+
+### string
+Make a string unicode if it isn’t already:
```

Comment:
Yes sorry. I initially copied all the Jinja descriptions when I ported it across to makes sure I had everything and a general gist of what it did. I've updated it to be more in line with JS as suggested.